### PR TITLE
feat(ui-map): add layers and districts french translations

### LINKED ISSUE #1274: Add layers and districts translations

### PR DESCRIPTION
**Pull request by hdinia**

### CODE REVIEW

File: webapp/public/locales/fr/main.json [hunk 1]
```diff
@@ -367,6 +367,18 @@
   "study.modelization.nodeProperties.filterByYear": "Sorties année par année",
   "study.modelization.nodeProperties.filterSynthesis": "Sorties de la synthèse",
   "study.modelization.map": "Carte",
+  "study.modelization.map.newArea": "Nouvelle zone",
+  "study.modelization.map.areaName": "Nom de la zone",
+  "study.modelization.map.layers": "Layers",
+  "study.modelization.map.layers.add": "Ajouter layer",
+  "study.modelization.map.layers.edit": "Modifier layers",
+  "study.modelization.map.layers.delete": "Supprimer layer",
+  "study.modelization.map.layers.question.deleteLayer": "Êtes-vous sûr de vouloir supprimer ce layer?",
```
Comment (hdinia) [inline webapp/public/locales/fr/main.json]: add space before "?"

File: webapp/public/locales/fr/main.json [hunk 2]
```diff
@@ -367,6 +367,18 @@
...
+  "study.modelization.map.districts": "Regroupement de zones",
+  "study.modelization.map.districts.add": "Ajouter regroupement de zones",
+  "study.modelization.map.districts.edit": "Modifier regroupements de zones",
+  "study.modelization.map.districts.delete": "Supprimer regroupement de zones",
+  "study.modelization.map.districts.question.deleteDistrict": "Êtes-vous sûr de vouloir supprimer ce regroupement de zones?",
```
Comment (hdinia) [inline webapp/public/locales/fr/main.json]: add space before "?"

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx [hunk 3]
```diff
@@ -115,14 +115,14 @@ function UpdateDistrictDialog(props: Props) {
...
+            label={t("global.outputs")}
             name="output"
-            label="Output"
             control={control}
             disabled={getValues("districtId") === ""}
             sx={{ ".MuiFormControlLabel-root": { m: 0 } }}
           />
           <StringFE
-            label="Comment"
+            label={t("global.comments")}
```
Comment (hdinia) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx]: rename comment ? (singular)

File: webapp/public/locales/en/main.json [hunk 4]
```diff
@@ -60,6 +60,8 @@
   "global.advancedParams": "Advanced parameters",
   "global.matrix": "Matrix",
   "global.matrices": "Matrices",
+  "global.outputs": "Outputs",
+  "global.comments": "Comments",
```
Comment (hdinia) [inline webapp/public/locales/en/main.json]: rename comment ?

File: webapp/public/locales/fr/main.json [hunk 5]
```diff
@@ -367,6 +366,19 @@
   "study.modelization.nodeProperties.filterByYear": "Sorties année par année",
   "study.modelization.nodeProperties.filterSynthesis": "Sorties de la synthèse",
   "study.modelization.map": "Carte",
+  "study.modelization.map.newArea": "Nouvelle zone",
+  "study.modelization.map.areaName": "Nom de la zone",
+  "study.modelization.map.layers": "Layers",
+  "study.modelization.map.layers.add": "Ajouter layer",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: "Ajouter un layer"

File: webapp/public/locales/fr/main.json [hunk 6]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.layers.edit": "Modifier layers",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: "Modifier les layers"

File: webapp/public/locales/fr/main.json [hunk 7]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.layers.delete": "Supprimer layer",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: Remove, use "global.delete" instead

File: webapp/public/locales/en/main.json [hunk 8]
```diff
@@ -367,6 +366,19 @@
   "study.modelization.nodeProperties.filterByYear": "Output year by year",
   "study.modelization.nodeProperties.filterSynthesis": "Synthesis outputs",
   "study.modelization.map": "Map",
+  "study.modelization.map.newArea": "New Area",
+  "study.modelization.map.areaName": "Area name",
+  "study.modelization.map.layers": "Layers",
+  "study.modelization.map.layers.add": "Add layer",
+  "study.modelization.map.layers.edit": "Edit layers",
+  "study.modelization.map.layers.delete": "Delete layer",
```
Comment (skamril) [inline webapp/public/locales/en/main.json]: Remove, use "global.delete" instead

File: webapp/public/locales/en/main.json [hunk 9]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.layers.question.deleteLayer": "Are you sure you want to delete this layer?",
```
Comment (skamril) [inline webapp/public/locales/en/main.json]: "Are you sure you want to delete '{{0}}' layer?"

File: webapp/public/locales/en/main.json [hunk 10]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.districts": "Districts",
+  "study.modelization.map.districts.field.comments": "Comments",
+  "study.modelization.map.districts.add": "Add district",
+  "study.modelization.map.districts.edit": "Edit districts",
+  "study.modelization.map.districts.delete": "Delete district",
```
Comment (skamril) [inline webapp/public/locales/en/main.json]: Remove, use "global.delete" instead

File: webapp/public/locales/en/main.json [hunk 11]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.districts.question.deleteDistrict": "Are you sure you want to delete this district?",
```
Comment (skamril) [inline webapp/public/locales/en/main.json]: "Are you sure you want to delete '{{0}}' district?"

File: webapp/public/locales/fr/main.json [hunk 12]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.layers.question.deleteLayer": "Êtes-vous sûr de vouloir supprimer ce layer ?",
+  "study.modelization.map.districts.field.comments": "Commentaires",
+  "study.modelization.map.districts": "Regroupement de zones",
+  "study.modelization.map.districts.add": "Ajouter regroupement de zones",
+  "study.modelization.map.districts.edit": "Modifier regroupements de zones",
+  "study.modelization.map.districts.delete": "Supprimer regroupement de zones",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: Remove, use "global.delete" instead

File: webapp/public/locales/fr/main.json [hunk 13]
```diff
@@ -367,6 +366,19 @@
   "study.modelization.nodeProperties.filterByYear": "Sorties année par année",
   "study.modelization.nodeProperties.filterSynthesis": "Sorties de la synthèse",
   "study.modelization.map": "Carte",
+  "study.modelization.map.newArea": "Nouvelle zone",
+  "study.modelization.map.areaName": "Nom de la zone",
+  "study.modelization.map.layers": "Layers",
+  "study.modelization.map.layers.add": "Ajouter layer",
+  "study.modelization.map.layers.edit": "Modifier layers",
+  "study.modelization.map.layers.delete": "Supprimer layer",
+  "study.modelization.map.layers.question.deleteLayer": "Êtes-vous sûr de vouloir supprimer ce layer ?",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: "Êtes-vous sûr de vouloir supprimer le layer '{{0}}' ?"

File: webapp/public/locales/fr/main.json [hunk 14]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.districts.field.comments": "Commentaires",
+  "study.modelization.map.districts": "Regroupement de zones",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: Keep the word "district" for all keys

File: webapp/public/locales/fr/main.json [hunk 15]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.districts.add": "Ajouter regroupement de zones",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: "Ajouter un disctrict"

File: webapp/public/locales/fr/main.json [hunk 16]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.districts.edit": "Modifier regroupements de zones",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: "Modifier les districts"

File: webapp/public/locales/fr/main.json [hunk 17]
```diff
@@ -367,6 +366,19 @@
...
+  "study.modelization.map.districts.delete": "Supprimer regroupement de zones",
+  "study.modelization.map.districts.question.deleteDistrict": "Êtes-vous sûr de vouloir supprimer ce regroupement de zones ?",
```
Comment (skamril) [inline webapp/public/locales/fr/main.json]: "Êtes-vous sûr de vouloir supprimer le district '{{0}}' ?"

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx [hunk 18]
```diff
@@ -115,14 +115,14 @@ function UpdateDistrictDialog(props: Props) {
             sx={{ mx: 0 }}
           />
           <SwitchFE
+            label={t("global.outputs")}
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx]: Add "study.modelization.map.districts.field.outputs"

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx [hunk 19]
```diff
@@ -148,7 +148,9 @@ function UpdateDistrictDialog(props: Props) {
               alert="warning"
               open
             >
-              <Typography sx={{ p: 3 }}>Delete district ?</Typography>
+              <Typography sx={{ p: 3 }}>
+                {t("study.modelization.map.districts.question.deleteDistrict")}
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx]: Rename the key : study.modelization.map.district.delete.confirm

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Layers/UpdateLayerDialog.tsx [hunk 20]
```diff
@@ -137,7 +137,9 @@ function UpdateLayerDialog(props: Props) {
               alert="warning"
               open
             >
-              <Typography sx={{ p: 3 }}>Delete layer ?</Typography>
+              <Typography sx={{ p: 3 }}>
+                {t("study.modelization.map.layers.question.deleteLayer")}
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Layers/UpdateLayerDialog.tsx]: Rename the key : study.modelization.map.layer.delete.confirm

File: webapp/public/locales/en/main.json [hunk 21]
```diff
@@ -60,6 +60,7 @@
   "global.advancedParams": "Advanced parameters",
```
Comment (skamril) [inline webapp/public/locales/en/main.json]: For keys: "layer" and "district" in singular

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx [hunk 22]
```diff
@@ -148,7 +148,11 @@ function UpdateDistrictDialog(props: Props) {
               alert="warning"
               open
             >
-              <Typography sx={{ p: 3 }}>Delete district ?</Typography>
+              <Typography sx={{ p: 3 }}>
+                {t("study.modelization.map.districts.delete.confirm", [
+                  getValues("districtId"),
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Map/MapConfig/Districts/UpdateDistrictDialog.tsx]: Use `districtsById` to get the name like layer

**Review by skamril — CHANGES_REQUESTED** [review]

**Review by skamril — DISMISSED** [review]

**Review by skamril — APPROVED** [review]

**Pull request merged**